src/app/rpmostree-builtin-kargs.cxx

```diff
@@ -325,14 +330,21 @@ rpmostree_builtin_kargs (int            argc,
         opt_kernel_append_strings = empty_strv;
       if (!opt_kernel_delete_strings)
         opt_kernel_delete_strings = empty_strv;
+      if (!opt_kernel_append_if_missing_strings)
+        opt_kernel_append_if_missing_strings = empty_strv;
+      if (!opt_kernel_delete_if_present_strings)
+        opt_kernel_delete_if_present_strings = empty_strv;
+
 
       /* call the generearted dbus-function */
       if (!rpmostree_os_call_kernel_args_sync (os_proxy,
                                                old_kernel_arg_string,
                                                (const char* const*) opt_kernel_append_strings,
                                                (const char* const*) opt_kernel_replace_strings,
                                                (const char* const*) opt_kernel_delete_strings,
-                                               options,
+                                               (const char* const*) opt_kernel_append_if_missing_strings,
+                                               (const char* const*) opt_kernel_delete_if_present_strings,
```

Comment:
I mentioned this in https://github.com/coreos/rpm-ostree/pull/2714; we can't modify the D-Bus method signature now that it's stable.

Instead, what I was suggesting was to put it inside the options GVariant, the way it's done for `reboot`, `initiating-command-line`, and `lock-finalization` above. You can see the documentation for GVariant and GVariantDict in https://developer.gnome.org/glib/stable/glib-GVariant.html. There's an example of putting an array in a `GVariantDict` in `get_modifiers_variant` in `rpmostree-clientlib.cxx`.